src/main/java/com/mraof/minestuck/blockentity/machine/TotemLatheBlockEntity.java

```diff
@@ -0,0 +1,441 @@
+	{
+		BlockState currentState = getLevel().getBlockState(getBlockPos());
+		if(!isBroken())
+		{
+			checkStates();
+			if(isBroken())
+				LOGGER.warn("Failed to notice a block being broken or misplaced at the totem lathe at {}", getBlockPos());
+		}
+		
+		if(!state.getValue(TotemLatheBlock.FACING).equals(currentState.getValue(TotemLatheBlock.FACING)))
+			return false;
+		return !isBroken();
+	}
+	
+	public void checkStates()
+	{
+		if(isBroken())
+			return;
+		
+		if(MSBlocks.TOTEM_LATHE.isInvalidFromSlot(level, getBlockPos()))
+			setBroken();
+	}
+	
+	private void dropItem(boolean inBlock, BlockPos pos, ItemStack stack)
+	{
+		Direction direction = getFacing();
+		BlockPos dropPos;
+		if(inBlock)
+			dropPos = pos;
+		else if(!Block.canSupportCenter(level, pos.relative(direction), direction.getOpposite()))
+			dropPos = pos.relative(direction);
+		else dropPos = pos;
+		
+		Containers.dropItemStack(level, dropPos.getX(), dropPos.getY(), dropPos.getZ(), stack);
+	}
+	
+	@Override
+	public void load(CompoundTag nbt)
+	{
+		super.load(nbt);
+		broken = nbt.getBoolean("broken");
+		card1 = ItemStack.of(nbt.getCompound("card1"));
+		card2 = ItemStack.of(nbt.getCompound("card2"));
+		isProcessing = nbt.getBoolean("isProcessing");
+		if(card1.isEmpty() && !card2.isEmpty())
+		{
+			card1 = card2;
+			card2 = ItemStack.EMPTY;
+		}
+	}
+	
+	@Override
+	public void saveAdditional(CompoundTag compound)
+	{
+		super.saveAdditional(compound);
+		compound.putBoolean("broken",broken);
+		compound.put("card1", card1.save(new CompoundTag()));
+		compound.put("card2", card2.save(new CompoundTag()));
+		compound.putBoolean("isProcessing", isProcessing);
+	}
+	
+	public void processContents()
+	{
+		ItemStack dowel = getDowel();
+		ItemStack output;
+		boolean success = false;
+		if(!dowel.isEmpty() && !AlchemyHelper.hasDecodedItem(dowel) && (!card1.isEmpty() || !card2.isEmpty()))
+		{
+			if(!card1.isEmpty() && !card2.isEmpty())
+				if(!AlchemyHelper.isPunchedCard(card1) || !AlchemyHelper.isPunchedCard(card2))
+					output = new ItemStack(MSItems.GENERIC_OBJECT.get());
+				else output = CombinationRecipe.findResult(new CombinerWrapper(card1, card2, CombinationMode.AND), level);
+			else
+			{
+				ItemStack input = card1.isEmpty() ? card2 : card1;
+				if(!AlchemyHelper.isPunchedCard(input))
+					output = new ItemStack(MSItems.GENERIC_OBJECT.get());
+				else output = AlchemyHelper.getDecodedItem(input);
+			}
+			
+			if(!output.isEmpty())
+				success = setCarvedItem(output);
+		}
+		
+		effects(success);
+	}
+	
+	public static void tick(Level level, BlockPos pos, BlockState state, TotemLatheBlockEntity blockEntity)
+	{
+		if(blockEntity.animationticks > 0)
+		{
+			blockEntity.animationticks--;
+			if(blockEntity.animationticks <= 0)
+			{
+				processContents(); //TODO in 1.16 there was the ITickableTileEntity, which had a nonstatic function. Now we use a static function and cannot handle processContents the same
+			}
+		}
+	}
```

Comment:
Regarding the TODO in this method, im unsure if I should change processContents to adjust(giving it some parameters)

Reply:
Adjusted it so that processContents is also static